htdocs/comm/propal/class/propal.class.php

```diff
@@ -1138,7 +1138,7 @@ public function create($user, $notrigger = 0)
 					$sql = "UPDATE ".MAIN_DB_PREFIX."propal";
 					$sql.= " SET fk_delivery_address = ".$this->fk_delivery_address;
 					$sql.= " WHERE ref = '".$this->db->escape($this->ref)."'";
-					$sql.= " AND entity = ".$conf->entity;
+					$sql.= " AND entity = ".$this->entity;
```

Comment:
This means you can create a proposal on entity 2 when your are logged on entity 1 ?

Reply:
@eldy 
yes that's the goal, but there are user rights in the Multicompany module. It is a request of a lot of customers

Reply:
You shoud introduce a
$sql.= " AND entity IN (".getEntity($this).")";    (to define entity allowed in write, compared to getEntites to know list en entities allowed in read)
so the module will have control on what is the entity to use and will be able to check permissions are ok. The $conf->entity will be returned by default. Without a centralized part of code controlled by the module, any simple api call that set the entity will be able to bypass any permissions.

Reply:
@eldy 
maybe "setEntity()" instead "getEntity" ?
(getEntity is already used)